src/openexr.imageio/exrinput_c.cpp

```diff
@@ -0,0 +1,1576 @@
+    //KDTDISABLE     errorf("Could not open file \"%s\"", name);
+    //KDTDISABLE     return false;
+    //KDTDISABLE }
+    //KDTDISABLE if (!valid_file(name, m_io)) {
+    //KDTDISABLE     errorf("\"%s\" is not an OpenEXR file", name);
+    //KDTDISABLE     return false;
+    //KDTDISABLE }
+
+    // Check any other configuration hints
+
+    // "missingcolor" gives fill color for missing scanlines or tiles.
+    if (const ParamValue* m = config.find_attribute("oiio:missingcolor")) {
+        if (m->type().basetype == TypeDesc::STRING) {
+            // missingcolor as string
+            m_missingcolor = Strutil::extract_from_list_string<float>(
+                m->get_string());
+        } else {
+            // missingcolor as numeric array
+            int n = m->type().basevalues();
+            m_missingcolor.clear();
+            m_missingcolor.reserve(n);
+            for (int i = 0; i < n; ++i)
+                m_missingcolor[i] = m->get_float(i);
+        }
+    } else {
+        // If not passed explicit, is there a global setting?
+        std::string mc = OIIO::get_string_attribute("missingcolor");
+        if (mc.size())
+            m_missingcolor = Strutil::extract_from_list_string<float>(mc);
+    }
+
+    // Clear the spec with default constructor
+    m_spec = ImageSpec();
+
+    // Establish an input stream. If we weren't given an IOProxy, create one
+    // now that just reads from the file.
+    try {
+        if (!m_userdata.m_io) {
+            m_userdata.m_io = new Filesystem::IOFile(name,
+                                                     Filesystem::IOProxy::Read);
+            m_local_io.reset(m_userdata.m_io);
+        }
+        OIIO_ASSERT(m_userdata.m_io);
+        if (m_userdata.m_io->mode() != Filesystem::IOProxy::Read) {
+            // If the proxy couldn't be opened in write mode, try to
+            // return an error.
+            std::string e = m_userdata.m_io->error();
+            errorf("Could not open \"%s\" (%s)", name,
+                   e.size() ? e : std::string("unknown error"));
+            return false;
+        }
+        m_userdata.m_io->seek(0);
+    } catch (const std::exception& e) {
+        m_local_io.reset();
+        m_userdata.m_io = nullptr;
+        errorf("OpenEXR exception: %s", e.what());
+        return false;
+    } catch (...) {  // catch-all for edge cases or compiler bugs
+        m_local_io.reset();
+        m_userdata.m_io = nullptr;
+        errorf("OpenEXR exception: unknown");
+        return false;
+    }
+
+    m_userdata.m_img                = this;
+    exr_context_initializer_t cinit = EXR_DEFAULT_CONTEXT_INITIALIZER;
+
+    cinit.error_handler_fn = &oiio_exr_error_handler;
+    cinit.user_data        = &m_userdata;
+    // KDT TODO: We could change the above to not always create an
+    // ioproxy? Although if we do that, we'll have to change the logic
+    // in the error handler where right now we assume that only
+    // happens in the valid file check
```

Comment:
yeah, that comment was around when the user does not provide a proxy - I like to think about less code, less bugs as a first course of action. But I suppose if the implementor changes the default behaviour of IOProxy, this will continue to leave this behaving consistently. Given I can remove the exception bits, this will look much simpler after